prototype2/js/model.js

```diff
@@ -51,14 +51,14 @@ class Model {
         let max = minmax[1];
         let diff = max - min;
         return function (d) {
-            return d > (min + diff * 7.0 / 8.0) ? colors[7] :
-                   d > (min + diff * 6.0 / 8.0) ? colors[6] :
-                   d > (min + diff * 5.0 / 8.0) ? colors[5] :
-                   d > (min + diff * 4.0 / 8.0) ? colors[4] :
-                   d > (min + diff * 3.0 / 8.0) ? colors[3] :
-                   d > (min + diff * 2.0 / 8.0) ? colors[2] :
-                   d > (min + diff * 1.0 / 8.0) ? colors[1] :
-                                                  colors[0];
+            var i = 7;
+            while (i >= 0) {
+                if (d >= (min + diff * i / 8.0)) {
+                    return colors[i];
+                } else {
+                    i--;
+                }
+            }

```

Comment:
Instead of looping, I think you can solve this directly. You'll have to use Math.floor().

You could also consider d3's quantize scale functions which would do a lot of edge case handling for you, but might require looking up quite a bit of documentation.

Reply:
I see what you mean now. I had updated the code accordingly.